File: experiments/java-sdk/examples/com/datadog/llmobs/experiments/examples/MinimalExperiment.java
package com.datadog.llmobs.experiments.examples;

import com.datadog.llmobs.experiments.Dataset;
import com.datadog.llmobs.experiments.Experiment;
import com.datadog.llmobs.experiments.ExperimentResult;
import com.datadog.llmobs.experiments.ExperimentsClient;

import java.util.Map;

/**
 * Smallest end-to-end experiment: one record, one evaluator, one URL printed. Use this as
 * a copy-paste starting point. Other examples in this directory add evaluators, metadata,
 * multi-record datasets, and dataset round-tripping on top of this skeleton.
 *
 * <p>Run with: {@code ./gradlew runMinimalExperiment}
 */
public final class MinimalExperiment {

    public static void main(String[] args) {
        ExperimentsClient client = ExperimentsClient.builder()
            .apiKey(System.getenv("DD_API_KEY"))
            .applicationKey(System.getenv("DD_APPLICATION_KEY"))
            .site(System.getenv().getOrDefault("DD_SITE", "datadoghq.com"))
            .projectName("java-sdk-minimal")
            .build();

        Dataset dataset = client.createDataset("minimal-" + System.currentTimeMillis())
            .addRecord(Map.of("name", "World"), "Hello, World");

        ExperimentResult result = Experiment.<Map<String, Object>, String>builder(client)
            .name("minimal-experiment")
            .dataset(dataset)
            .task((input, config) -> "Hello, " + input.get("name"))
            .evaluator("exact_match", (in, out, expected) -> out.equals(expected))
            .build()
            .run();

        System.out.println("Experiment URL: " + result.url());
    }
}

File: experiments/node-sdk/examples/minimalExperiment.ts
/**
 * MinimalExperiment — the smallest end-to-end run: a single record, one task,
 * one evaluator. Writes to the `node-sdk-minimal` project.
 *
 *   DD_API_KEY=... DD_APPLICATION_KEY=... npm run runMinimalExperiment
 */
import { Experiment, ExperimentsClient } from "../src/index.js";

function requireEnv(name: string): string {
  const v = process.env[name];
  if (!v || v.trim() === "") {
    console.error(`Missing required env var: ${name}`);
    process.exit(1);
  }
  return v;
}

type Input = { name: string };

async function main(): Promise<void> {
  const client = new ExperimentsClient({
    apiKey: requireEnv("DD_API_KEY"),
    applicationKey: requireEnv("DD_APPLICATION_KEY"),
    site: process.env.DD_SITE ?? "datadoghq.com",
    projectName: "node-sdk-minimal",
  });

  const dataset = client
    .createDataset(`minimal-${Date.now()}`)
    .addRecord({ name: "World" }, "Hello, World");

  const result = await Experiment.builder<Input, string>(client)
    .name("minimal-experiment")
    .dataset(dataset)
    .task((input) => `Hello, ${input.name}`)
    .evaluator("exact_match", (_input, output, expected) => output === expected)
    .build()
    .run();

  console.log(`Dataset URL    : ${dataset.url()}`);
  console.log(`Experiment URL : ${result.url}`);
  console.log(`Experiment ID  : ${result.experimentId}`);
  console.log(`Rows           : ${result.rows.length}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
